packages/medusa/src/repositories/customer-group.ts

```diff
@@ -1,5 +1,64 @@
-import { EntityRepository, Repository } from "typeorm"
+import {
+  EntityRepository,
+  Repository,
+  getManager,
+  getConnection,
+} from "typeorm"
+import { MedusaError } from "medusa-core-utils"
 import { CustomerGroup } from "../models/customer-group"
+import { CustomerGroupsBatchCustomer } from "../types/customer-groups"
+import { Customer } from ".."
 
 @EntityRepository(CustomerGroup)
-export class CustomerGroupRepository extends Repository<CustomerGroup> {}
+export class CustomerGroupRepository extends Repository<CustomerGroup> {
+  async addCustomerBatch(
+    groupId: string,
+    customerIds: CustomerGroupsBatchCustomer[]
+  ): Promise<CustomerGroup> {
+    try {
+      const customerGroup = await this.findOne(groupId)
+      if (!customerGroup) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_FOUND,
+          `CustomerGroup with id ${groupId} was not found`
+        )
+      }
+
```

Comment:
You could move this into the catch as well since the foreign key constraint on group will also throw 23503.